cmd/pod-scaler/frontend/src/app/store.ts

```diff
@@ -0,0 +1,70 @@
+import { configureStore, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { TreeViewDataItem } from '@patternfly/react-core';
+
+const initialTrees: Record<string, TreeViewDataItem[]> = {};
+
+export const trees = createSlice({
+  name: 'trees',
+  initialState: initialTrees,
+  reducers: {
+    updateTrees: (state, action: PayloadAction<Record<string, TreeViewDataItem[]>>) => {
+      for (const name in action.payload) {
+        if (action.payload.hasOwnProperty(name)) {
+          state[name] = action.payload[name];
```

Comment:
You don't want to update state directly in a reducer. You want to return the new state without mutating the original state.

Reply:
I think the docs for the particular library being used here talk at lengh about how they do some magic behind the scenes to hide this implementation detail and make it more idiomatic - update here is on some copy that the wrapper returns to redux itself.